sw/device/lib/testing/test_framework/freertos_port.S

```diff
@@ -0,0 +1,162 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+// Ibex does not implement additional registers beyond the RV32I spec.
+#define PORT_WORD_SIZE 4
+#define PORT_CONTEXT_SIZE (30 * PORT_WORD_SIZE)
+
+.extern pxCurrentTCB
+.extern xISRStackTop
+
+// -----------------------------------------------------------------------------
+
+/**
+ * FreeRTOS, expects this function to exist and uses it to start the first task. 
+ */
+.balign 8
+.global xPortStartFirstTask
+.type xPortStartFirstTask, @function
+xPortStartFirstTask:
+
+  // Load the stack pointer for the current TCB (just going to clobber sp here
+  // since we are setting it here anyway).
+  lw  sp, pxCurrentTCB
+  lw  sp, 0(sp)
+
+  // NOTE: for starting the FreeRTOS scheduler, the exception return address is
+  // used as the function return address. See pxPortInitialiseStack below.
+  lw  ra, 0(sp)
+
+  // Restore registers initialized on task start.
+  lw   t1,  3 * PORT_WORD_SIZE(sp)
+  lw   t2,  4 * PORT_WORD_SIZE(sp)
+  lw   s0,  5 * PORT_WORD_SIZE(sp)
+  lw   s1,  6 * PORT_WORD_SIZE(sp)
+  lw   a0,  7 * PORT_WORD_SIZE(sp) // task parameters (pvParameters)
+  lw   a1,  8 * PORT_WORD_SIZE(sp)
+  lw   a2,  9 * PORT_WORD_SIZE(sp)
+  lw   a3, 10 * PORT_WORD_SIZE(sp)
+  lw   a4, 11 * PORT_WORD_SIZE(sp)
+  lw   a5, 12 * PORT_WORD_SIZE(sp)
+  lw   a6, 13 * PORT_WORD_SIZE(sp)
+  lw   a7, 14 * PORT_WORD_SIZE(sp)
+  lw   s2, 15 * PORT_WORD_SIZE(sp)
+  lw   s3, 16 * PORT_WORD_SIZE(sp)
+  lw   s4, 17 * PORT_WORD_SIZE(sp)
+  lw   s5, 18 * PORT_WORD_SIZE(sp)
+  lw   s6, 19 * PORT_WORD_SIZE(sp)
+  lw   s7, 20 * PORT_WORD_SIZE(sp)
+  lw   s8, 21 * PORT_WORD_SIZE(sp)
+  lw   s9, 22 * PORT_WORD_SIZE(sp)
+  lw  s10, 23 * PORT_WORD_SIZE(sp)
+  lw  s11, 24 * PORT_WORD_SIZE(sp)
+  lw   t3, 25 * PORT_WORD_SIZE(sp)
+  lw   t4, 26 * PORT_WORD_SIZE(sp)
+  lw   t5, 27 * PORT_WORD_SIZE(sp)
+  lw   t6, 28 * PORT_WORD_SIZE(sp)
+
+  // Initialize t0 to the value of MSTATUS with global interrupts enabled, which
+  // is required because this returns with ret, not eret.
+  lw t0, 29 * PORT_WORD_SIZE(sp)
+  addi t0, t0, 0x08
```

Comment:
We probably want to use `ori` to set this bit. Also, for bitfields I would tend to use a shift to generate the immediate (since the shift amount matches the index in the spec) but it doesn't really matter:
```
ori t0, t0, 1<<3 // set MIE field
```

Reply:
Fixed.